docs/signing/local-signing.md

```diff
@@ -0,0 +1,161 @@
+
+The rest of this tutorial uses OpenSSL (a set of cryptographic utilities). If OpenSSL is not installed on your system, see [OpenSSL](https://www.openssl.org/source/) for the source distribution or the [list of unofficial binary distributions](https://wiki.openssl.org/index.php/Binaries).
+
+### 2. Extract credentials
+
+To work with the certificate, you need to extract it. When the CAI SDK adds Content Credentials to an asset, it incorporates the certificate (including the associated public key) into the manifest.
+Use the commands below to extract the key and certificate chain. If prompted, enter the password that was used to generate the `.pfx` file.
+
+:::tip
+Make sure you are using a recent version of OpenSSL.
+:::
+
+#### Troubleshooting errors
+
+In this step, OpenSSL may report errors when extracting the key or certificate chain.  In many cases, if OpenSSL generates the output file, you can ignore the messages.  
+
+For example, the following error message means the `.pfx` was encrypted with an older standard:
+
+```
+Shrouded Keybag: pbeWithSHA1And3- KeyTripleDES-CBC, Iteration 2000
+PKCS7 Encrypted data: pbeWithSHA1And40BitRC2- CBC, Iteration 2000
+Error outputting keys and certificates
+```
+
+#### Extract the key
+
+```shell
+openssl pkcs12 -in mycertfile.pfx -nocerts -out mykey.pem -nodes
+```
+
+:::tip
+Check to make sure the above command generated a `.pem` file and it's not an empty file.  For more information, see [Troubleshooting errors](#troubleshooting-errors) above.
+:::
+
+#### Extract certificate chain
+
+For many certificate providers, the `.pfx` file contains not just your certificate but the complete certificate trust chain, which you can extract with a command like this:
+
+```shell
+openssl pkcs12 -in mycertfile.pfx -nokeys -out mycerts.pub
+```
+
+When the `.pfx` file does not contain the certificate chain, you can obtain it from your provider.
+
+### 3. Determine signature algorithm
+
+To use the credentials extracted above you must know the signature types they support. Typically, you'll already know this information or the certificate provider will provide it. 
+
+You can also enter this OpenSSL command to dump information about the public key used with the certificate:
+
+```shell
+openssl x509 -inform PEM -in mycerts.pub -text
+```
+
+Where `mycerts.pub` is the file containing the certificate chain from signing certificate to the last certificate before the root CA, concatenated.
+
+This command produces a text summary of the certificate properties, as shown in the example below. Look for a line containing `Signature Algorithm`. The public key indicates the signature algorithm used. See the table in [Getting a certificate](get-cert.md#signature-types) to determine the corresponding signature type.
+
+For this example with a certificate issued by GlobalSign, `Signature Algorithm: sha256WithRSAEncryption` corresponds to the PS256 signature type.
```

Comment:
`Signature Algorithm: sha256WithRSAEncryption` 

should be
'Public Key Algorithm: rsassaPss' with PSS parameter 'Hash Algorithm: SHA2-256'

Reply:
I fixed that already as well.